src/com/teamcautionrobotics/robot2018/Lift.java

```diff
@@ -0,0 +1,161 @@
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.wpilibj.PIDOutput;
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.PIDSourceType;
+
+public class Lift implements PIDOutput, PIDSource {
+
+    enum LiftLevel {
+        GROUND(0), SWITCH(19), LOW_SCALE(48), HIGH_SCALE(76);
+
+        private static LiftLevel[] values = values();
+
+        double height;
+
+        private LiftLevel(double height) {
+            this.height = height;
+        }
+
+        public LiftLevel next() {
+            if (this.ordinal() == values.length - 1) {
+                return values[this.ordinal()];
+            } else {
+                return values[this.ordinal() + 1];
+            }
+        }
+
+        public LiftLevel previous() {
+            if (this.ordinal() == 0) {
+                return values[this.ordinal()];
+            } else {
+                return values[this.ordinal() - 1];
+            }
+        }
+    }
+
+    private VictorSP liftMotor;
+    private Encoder liftEncoder;
+    private PIDController pidController;
+
+    private double destinationHeight;
+
+    public Lift(int motorPort, int encoderChannelA, int encoderChannelB, double Kp, double Ki,
+            double Kd) {
+        liftMotor = new VictorSP(motorPort);
+        liftEncoder = new Encoder(encoderChannelA, encoderChannelB);
+        liftEncoder.setDistancePerPulse((4 * Math.PI) / 1024);
+        pidController = new PIDController(Kp, Ki, Kd, 0, this, this);
+        pidController.setOutputRange(-1, 1);
+        pidController.setAbsoluteTolerance(3);
+    }
+
+    /**
+     * @param power positive is ascending, negative is descending, range of [-1, 1]
+     */
+    public void move(double power) {
+        double rateOfChange = power * 10; //inches per second
+        double dt = 0.02; //seconds
+        pidController.setSetpoint(getCurrentHeight() + rateOfChange * dt);
+    }
+
+    public void ascend() {
+        this.move(1.0);
+    }
+
+    public void descend() {
+        this.move(-1.0);
+    }
+
+    public void stop() {
+        this.move(0);
+    }
+
+    public void setLevel(LiftLevel destinationLiftLevel) {
+        destinationHeight = destinationLiftLevel.height;
+        pidController.setSetpoint(destinationHeight);
+        enablePID();
+    }
+
+    public void setHeight(double height) {
+        pidController.setSetpoint(height);
+        enablePID();
+    }
+
+    public LiftLevel getCurrentLiftLevel() {
+        return convertHeightToLiftLevel(getCurrentHeight());
+    }
+
+    public double getCurrentHeight() {
+        return getDistance();
+    }
+
+    public double getDestinationHeight() {
+        return pidController.getSetpoint();
+    }
+
+    public LiftLevel getDestinationLiftLevel() {
+        return convertHeightToLiftLevel(getDestinationHeight());
+    }
+
+    private LiftLevel convertHeightToLiftLevel(double height) {
```

Comment:
This should probably be `public static`. Also `LiftLevelFromHeight` may be a better name.